### PR TITLE
#158531993 Add tests for the endpoints

### PR DESCRIPTION
**Pull request by meshack-mbuvi**
#### What does this PR do?
Have Create,Retrieve on resources endpoint in Ride-my-way api
#### Description of Task to be completed?
Adds unittests for:
```
- GET /rides  Fetch all ride offers  
- GET /rides/<rideId>  Fetch a single ride offer 
- POST /rides  Create a ride offer 
- POST /rides/<rideId>/requests  
```
#### How should this be manually tested?
using your terminal,
- clone this repo
- `cd ride-my-way-api`
- set and activate your virtual environment
- `git checkout ft-api-tests-158531993`
- Run `nosetests`

#### Any background context you want to provide?
Unit testing gives peace of mind as most of the bugs are identified and corrected at early stages of product development.
#### What are the relevant pivotal tracker stories?
[[#158531993]](https://www.pivotaltracker.com/story/show/158531993)

### CODE REVIEW

File: tests/test_rides.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        self.app = create_app('testing')
+        self.app.config.from_object(configuration['testing'])
+        self.app = self.app.test_client()
+
+    def tearDown(self):
+        """Release flask app instance"""
+        self.app = None
+
+    def test_create_ride(self):
+        """test user can get available rides"""
```
Comment (JoshuaOndieki) [inline tests/test_rides.py]: Update the *test_create_ride* Docstring

File: tests/test_rides.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        """Assumes no ride with a negative id number"""
+        response = self.app.get('/api/v1/rides/-1',
+                                content_type='application/json')
+        self.assertEqual(response.status_code, 404)
+        response_data = json.loads(response.get_data().decode('utf-8'))
+        self.assertEqual(len(response_data), 0)
+
+    def test_post_join_ride(self):
+        """test user can get available rides"""
+        response = self.app.get('/api/v1/rides/1/requests',
```
Comment (JoshuaOndieki) [inline tests/test_rides.py]: Is joining a ride a *GET* or *POST*?

File: tests/test_rides.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+                                content_type='application/json')
+        """"message":"."""
+        self.assertEqual(response.status_code, 200)
+        response_data = json.loads(response.get_data().decode('utf-8'))
+        self.assertEqual(response_data['message'],
+                         "Your request has been send.")
+
+    def test_post_join_non_existing_ride_fails(self):
+        """test user can get available rides"""
+        response = self.app.get('/api/v1/rides/-1/requests',
```
Comment (JoshuaOndieki) [inline tests/test_rides.py]: Is joining a ride a GET or POST?

Comment (meshack-mbuvi) [reply tests/test_rides.py]: it is a post man

Comment (JoshuaOndieki) [reply tests/test_rides.py]: You are performing get requests

Comment (meshack-mbuvi) [reply tests/test_rides.py]: Tests update already

**Pull request merged**